docs/dymos_book/examples/balanced_field/balanced_field_funccomp.ipynb

```diff
@@ -0,0 +1,590 @@
+    "rto.set_time_options(fix_initial=False, duration_bounds=(1, 1000), duration_ref=1.0)\n",
+    "rto.add_state('r', fix_initial=False, lower=0, ref=1000.0, defect_ref=1000.0)\n",
+    "rto.add_state('v', fix_initial=False, lower=0, ref=100.0, defect_ref=100.0)\n",
+    "rto.add_parameter('alpha', val=0.0, opt=False, units='deg')\n",
+    "rto.add_timeseries_output('*')\n",
+    "\n",
+    "# Third Phase: V1 to Vr - single engine operable\n",
+    "v1_to_vr = dm.Phase(ode_class=BalancedFieldODEComp, transcription=dm.Radau(num_segments=3),\n",
+    "                    ode_init_kwargs={'mode': 'runway'})\n",
+    "v1_to_vr.set_time_options(fix_initial=False, duration_bounds=(1, 1000), duration_ref=1.0)\n",
+    "v1_to_vr.add_state('r', fix_initial=False, lower=0, ref=1000.0, defect_ref=1000.0)\n",
+    "v1_to_vr.add_state('v', fix_initial=False, lower=0, ref=100.0, defect_ref=100.0)\n",
+    "v1_to_vr.add_parameter('alpha', val=0.0, opt=False, units='deg')\n",
+    "v1_to_vr.add_timeseries_output('*')\n",
+    "\n",
+    "# Fourth Phase: Rotate - single engine operable\n",
+    "rotate = dm.Phase(ode_class=BalancedFieldODEComp, transcription=dm.Radau(num_segments=3),\n",
+    "                  ode_init_kwargs={'mode': 'runway'})\n",
+    "rotate.set_time_options(fix_initial=False, duration_bounds=(1.0, 5), duration_ref=1.0)\n",
+    "rotate.add_state('r', fix_initial=False, lower=0, ref=1000.0, defect_ref=1000.0)\n",
+    "rotate.add_state('v', fix_initial=False, lower=0, ref=100.0, defect_ref=100.0)\n",
+    "rotate.add_polynomial_control('alpha', order=1, opt=True, units='deg', lower=0, upper=10, ref=10, val=[0, 10])\n",
+    "rotate.add_timeseries_output('*')\n",
+    "\n",
+    "# Fifth Phase: Climb to target speed and altitude at end of runway.\n",
+    "climb = dm.Phase(ode_class=BalancedFieldODEComp, transcription=dm.Radau(num_segments=5),\n",
+    "                 ode_init_kwargs={'mode': 'climb'})\n",
+    "climb.set_time_options(fix_initial=False, duration_bounds=(1, 100), duration_ref=1.0)\n",
+    "climb.add_state('r', fix_initial=False, lower=0, ref=1000.0, defect_ref=1000.0)\n",
+    "climb.add_state('h', fix_initial=True, lower=0, ref=1.0, defect_ref=1.0)\n",
+    "climb.add_state('v', fix_initial=False, lower=0, ref=100.0, defect_ref=100.0)\n",
+    "climb.add_state('gam', fix_initial=True, lower=0, ref=0.05, defect_ref=0.05)\n",
+    "climb.add_control('alpha', opt=True, units='deg', lower=-10, upper=15, ref=10)\n",
+    "climb.add_timeseries_output('*')\n",
+    "\n",
+    "# Instantiate the trajectory and add phases\n",
+    "traj = dm.Trajectory()\n",
+    "p.model.add_subsystem('traj', traj)\n",
+    "traj.add_phase('br_to_v1', br_to_v1)\n",
+    "traj.add_phase('rto', rto)\n",
+    "traj.add_phase('v1_to_vr', v1_to_vr)\n",
+    "traj.add_phase('rotate', rotate)\n",
+    "traj.add_phase('climb', climb)\n",
+    "\n",
+    "# Add parameters common to multiple phases to the trajectory\n",
```

Comment:
This is ALOT of paramters. Most of them are things you probably would not change... like mu.

can we use default values for these in the function arguments (hopefully the function comp API picks those up for default values)? Then just promote some of the inputs to demonstrate you can and add a comment that others are left at their default values? 

As is this code makes it seem like you *must* promote everything as prameters.

Reply:
I took a pass at removing some of the parameters that didn't change from phase to phase.